File: julia/ExtendedEuclidean.jl
#=
    Created by Munawar Hasan
=#

function xgcd(a, b)
    s = 0
    r = b
    old_s = 1
    old_r = a

    while r != 0
        quotient = old_r ÷ r

        temp = r
        r = old_r - quotient * r
        old_r = temp

        temp = s
        s = old_s - quotient * s
        old_s = temp
    end
    if b != 0
        bezout_t = (old_r - old_s * a) ÷ b
    else
        bezout_t = 0
    end
    println("Bézout coefficients: ", (old_s, bezout_t))
end

xgcd(2304, 13)

File: rust/ExtendedEuclidean.rs
/*
    Created by Munawar Hasan
*/

fn main() {
    xgcd(2304, 13);
}

fn xgcd(a: i64, b: i64){

    let mut s = 0i64; 
    let mut r = b;
    let mut old_s = 1i64;
    let mut old_r = a;
    let bezout_t: i64;
    
    while r != 0{
        let quotient = old_r / r;
        
        let mut temp = r;
        
        r = old_r - quotient * r;
        old_r = temp;

        temp = s;

        s = old_s - quotient * s;
        old_s = temp;
    }
    if b != 0{
        bezout_t = (old_r - old_s * a) / b;
    }
    else{
        bezout_t = 0;
    }
    println!("Bézout coefficients: ({}, {})", old_s, bezout_t);
}